java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/SearchService.java

```diff
@@ -361,6 +383,16 @@ public Map<String, List<String>> searchPoiCategories() {
         return res;
     }
     
+    private MapPoiTypes getMapPoiTypes(String locale) throws XmlPullParserException, IOException {
+        MapPoiTypes mapPoiTypes = MapPoiTypes.getDefault();
+        String localPath = locale.equals("en") ? "values" : "values-" + locale;
+        Map<String, String> phrases = Algorithms.parseStringsXml(new File(andTranslationsLocation + localPath + "/phrases.xml"));
```

Comment:
dangerous File access (externally-defined by locale string). should be sanitized. at minimum, remove '/' chars from `locale`